plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -486,6 +481,38 @@ protected TableSnapshot getSnapshot(ConnectorSession session, SchemaTableName ta
         }
     }
 
+    private static long getVersionNumberFromConnectorTableVersion(TrinoFileSystem fileSystem, String tableLocation, ConnectorTableVersion version)
+    {
+        return switch (version.getPointerType()) {
+            // TODO https://github.com/trinodb/trino/issues/21024 Add support for reading tables with temporal versions
+            case TEMPORAL -> throw new TrinoException(NOT_SUPPORTED, "This connector does not support reading tables with TIMESTAMP AS OF");
+            case TARGET_ID -> getTargetVersionNumberFromVersion(fileSystem, tableLocation, version);
+        };
+    }
+
+    private static long getTargetVersionNumberFromVersion(TrinoFileSystem fileSystem, String tableLocation, ConnectorTableVersion version)
+    {
+        long snapshotId;
+        if (version.getVersionType() == SMALLINT || version.getVersionType() == TINYINT || version.getVersionType() == INTEGER || version.getVersionType() == BIGINT) {
+            snapshotId = (long) version.getVersion();
```

Comment:
can we instead do `if (version.getVersion() instanceof Long snapshotId) { return snapshotId; }` ?

Reply:
The logic will break when we support time travel with DATE type.

Reply:
Can we verify and throw in that case?

Reply:
I think allowing specific types (only 3) is better than denying unsupported types (date, time, interval, real and etc) in this case.